### PR TITLE
OCPBUGS-16334: Clarify updateActuatorConditions & update lastSync

### PR DESCRIPTION
**Pull request by bentito**
For `updateActuatorConditions()` usage, see inline comments please, basically this PR just cleans up the function to hopefully show conditions are updated properly.
 
`cr.Status.LastSyncTimestamp` is updated now.

### CODE REVIEW

File: pkg/operator/credentialsrequest/credentialsrequest_controller.go
Comment (bentito) [inline pkg/operator/credentialsrequest/credentialsrequest_controller.go:1]: https://github.com/openshift/cloud-credential-operator/pull/568/files#diff-ba64ecf2acf81969711010121ce3acdef206490c0d5aa1b9ee03a51321c6bdccR713 we're calling `updateActuatorConditions`. This PR cleans up that function and hopefully makes it clear we are setting the actuator conditions properly?

Comment (fxierh) [reply pkg/operator/credentialsrequest/credentialsrequest_controller.go:1]: Sorry for the confusion in forum-cloud-credential-operator. 

What I meant is, after CR.status.conditions are properly set by `updateActuatorConditions`, we should IMO still call r.Client.Status().Update(ctx, cr) after that.

File: pkg/operator/credentialsrequest/credentialsrequest_controller.go [hunk 1]
```diff
@@ -711,6 +711,11 @@ func (r *ReconcileCredentialsRequest) Reconcile(ctx context.Context, request rec
 
 				logger.Errorf("errored with condition: %v", t.Reason())
 				r.updateActuatorConditions(cr, t.Reason(), syncErr)
+				// Update the status of the CredentialsRequest object
+				err := r.Client.Status().Update(ctx, cr)
```
Comment (bentito) [inline pkg/operator/credentialsrequest/credentialsrequest_controller.go:715]: @fxierh This is where you meant to call Update, right?

File: pkg/operator/credentialsrequest/credentialsrequest_controller.go [hunk 2]
```diff
@@ -831,30 +836,30 @@ func (r *ReconcileCredentialsRequest) CreateOrUpdateOnCredsExist(ctx context.Con
 }
 
 func (r *ReconcileCredentialsRequest) updateActuatorConditions(cr *minterv1.CredentialsRequest, reason minterv1.CredentialsRequestConditionType, conditionError error) {
+	// Clear all existing conditions first
+	clearAllConditions(cr)
```
Comment (abutcher) [inline pkg/operator/credentialsrequest/credentialsrequest_controller.go:840]: I think we should revert to the original logic in `updateActuatorConditions`. The outcome of this switch case will be the same in terms of which conditions are set but we'll lose fields like `lastTransitionTime` should a condition remain the same.

File: pkg/operator/credentialsrequest/credentialsrequest_controller.go (hunk 1 above)
Comment (fxierh) [reply pkg/operator/credentialsrequest/credentialsrequest_controller.go:715]: Yep

File: pkg/operator/credentialsrequest/credentialsrequest_controller.go (hunk 2 above)
Comment (bentito) [reply pkg/operator/credentialsrequest/credentialsrequest_controller.go]: k reverting logic there

**Comment by abutcher** [comment]
Going to need a bug card for this one.

**Comment by bentito** [comment]
closing and reopening to hopefully get e2e-hypershift test unstuck.

**Pull request merged**